Sources/Apollo/HTTPNetworkTransport.swift

```diff
@@ -105,106 +112,166 @@ public class HTTPNetworkTransport: NetworkTransport {
           throw GraphQLHTTPResponseError(body: data, response: httpResponse, kind: .invalidResponse)
         }
-        let response = GraphQLResponse(operation: operation, body: body)
-        completionHandler(response, nil)
+        if self._enableAutoPersistedQueries,
+          let error = body["errors"] as? [JSONObject],
+          let errorMsg = error.filter ({ $0["message"] as? String != nil }).first?["message"] as? String,
+          ["PersistedQueryNotFound","PersistedQueryNotSupported"].contains(errorMsg) {
+          
+          let errorKind: GraphQLHTTPResponseError.ErrorKind = (errorMsg == "PersistedQueryNotFound") ? .persistedQueryNotFound : .persistedQueryNotSupported
+          let apqsError = GraphQLHTTPResponseError(body: data,
+                                                   response: httpResponse,
+                                                   kind: errorKind)
+          
+          let retryByDefault: Bool
+          if let reason = reason as? GraphQLHTTPResponseError,
+            reason.kind == apqsError.kind {
+            // if the same error occurs, respect retry option from delegate
+            retryByDefault = false
+          } else {
+            retryByDefault = true
+          }
+          // Auto Persisted Query handling
+          self.handleErrorOrRetry(operation: operation,
+                                  error: apqsError,
+                                  for: request,
+                                  response: response,
+                                  retryByDefault: retryByDefault,
+                                  completionHandler: completionHandler)
+
+        }else {
+          // no errors
+          let response = GraphQLResponse(operation: operation, body: body)
+          completionHandler(response, nil)
+        }
       } catch let parsingError {
-        self?.handleErrorOrRetry(operation: operation,
-                                 error: parsingError,
-                                 for: request,
-                                 response: response,
-                                 completionHandler: completionHandler)
+        self.handleErrorOrRetry(operation: operation,
+                                error: parsingError,
+                                for: request,
+                                response: response,
+                                completionHandler: completionHandler)
+        
       }
     }
-    
     task.resume()
-    
     return task
   }
-
-  private let sendOperationIdentifiers: Bool
   
+  // retryByDefault: still retry if delegate is absent, otherwise repect the value from delegate
   private func handleErrorOrRetry<Operation>(operation: Operation,
                                              error: Error,
                                              for request: URLRequest,
                                              response: URLResponse?,
+                                             retryByDefault: Bool = false,
```

Comment:
I think it might be better to keep this tied to the set value of APQ on the instance rather than passing it as a parameter - that'd make it considerably clearer where this is coming from.